pkg/operator/utils/topology.go

```diff
@@ -0,0 +1,22 @@
+package utils
+
+import (
+	opv1 "github.com/openshift/api/operator/v1"
+	"sort"
+)
+
+func GetTopologyCategories(clusterCSIDriver *opv1.ClusterCSIDriver) []string {
+	driverConfig := clusterCSIDriver.Spec.DriverConfig
+	if driverConfig.DriverType == opv1.VSphereDriverType {
+		vSphereConfig := driverConfig.VSphere
+		if vSphereConfig != nil && len(vSphereConfig.TopologyCategories) > 0 {
+			topologyCategories := vSphereConfig.TopologyCategories
+			// sort the topologyCategories so as they are stable and don't change their order
+			sort.Slice(topologyCategories, func(i, j int) bool {
+				return topologyCategories[i] < topologyCategories[j]
+			})
```

Comment:
Why is this sorted? Should `topology-categories` in the ini file have the same order of labels as this `TopologyCategories` field? IMO it would be quite confusing if the operator changes the order.

Reply:
The topology categories in API is an array whereas one in ini file is a single value string. It is possible that whenever we convert the array to the string, order may change and it could result in driver restart. That is why I am sorting them, so as to keep the order "stable".

From driver perspective, I don't think order in which categories are specified matters (I tested by changing the order and it works the same).

Reply:
> It is possible that whenever we convert the array to the string, order may change

How is it possible? It's just an array / slice and go does not reshuffle them randomly.

Reply:
Hmm, that is fair. I have removed sorting for now.